pkg/service/keystore/service_test.go

```diff
@@ -94,6 +94,72 @@ func TestEncryptDecryptAllKeyTypes(t *testing.T) {
 }
 
 func TestStoreAndGetKey(t *testing.T) {
+	keyStore, err := createKeyStoreService(t)
+	assert.NoError(t, err)
+	assert.NotEmpty(t, keyStore)
+
+	// store the key
+	_, privKey, err := crypto.GenerateEd25519Key()
+	assert.NoError(t, err)
+	err = keyStore.StoreKey(context.Background(), StoreKeyRequest{
+		ID:               "test-id",
+		Type:             crypto.Ed25519,
+		Controller:       "test-controller",
+		PrivateKeyBase58: base58.Encode(privKey),
+	})
+	assert.NoError(t, err)
+
+	// get it back
+	keyResponse, err := keyStore.GetKey(context.Background(), GetKeyRequest{ID: "test-id"})
+	assert.NoError(t, err)
+	assert.NotEmpty(t, keyResponse)
+	assert.Equal(t, privKey, keyResponse.Key)
+
+	// make sure can create a signer properly
+	signer, err := jwx.NewJWXSigner("test-id", "kid", keyResponse.Key)
+	assert.NoError(t, err)
+	assert.NotEmpty(t, signer)
+}
+
+func TestRevokeKey(t *testing.T) {
+	keyStore, err := createKeyStoreService(t)
+	assert.NoError(t, err)
+	assert.NotEmpty(t, keyStore)
+
+	// store the key
+	_, privKey, err := crypto.GenerateEd25519Key()
+	assert.NoError(t, err)
+	keyID := "test-revocation-id"
+	err = keyStore.StoreKey(context.Background(), StoreKeyRequest{
+		ID:               keyID,
+		Type:             crypto.Ed25519,
+		Controller:       "test-revocation-controller",
+		PrivateKeyBase58: base58.Encode(privKey),
+	})
+	assert.NoError(t, err)
+
+	// get it back
+	keyResponse, err := keyStore.GetKey(context.Background(), GetKeyRequest{ID: keyID})
+	assert.NoError(t, err)
+	assert.NotEmpty(t, keyResponse)
+	assert.Equal(t, privKey, keyResponse.Key)
+	assert.Equal(t, false, keyResponse.Revoked)
+	assert.Empty(t, keyResponse.RevokedAt)
+
+	// revoke the key
+	err = keyStore.RevokeKey(context.Background(), RevokeKeyRequest{ID: keyID})
+	assert.NoError(t, err)
+
+	// get the key after revocation
+	keyResponse, err = keyStore.GetKey(context.Background(), GetKeyRequest{ID: keyID})
+	assert.NoError(t, err)
+	assert.NotEmpty(t, keyResponse)
+	assert.Equal(t, privKey, keyResponse.Key)
+	assert.Equal(t, true, keyResponse.Revoked)
+	assert.NotEmpty(t, keyResponse.RevokedAt)
```

Comment:
I have tried to do it, but stopped after discovering that the keystore service does not use `Clock`, it uses `time.Now()` instead. I noticed that all other services also use `time.Now()` with the only exception being the manifest service that uses `Clock`. However, the [Clock module](https://github.com/benbjohnson/clock) is unmaintained. Are you sure?

Reply:
Thanks for pointing out the Clock is now archived. I wasn't aware of that! That said, I think it's still a good approach so that times are tested.

I'm make a ticket to clean up other places that use `time.Now()`.

Reply:
Could you elaborate on how the times should be tested within this PR specifically?

Reply:
Following the same pattern as manifest service. 
1. Create a field of type `Clock` in the `Service` struct.
2. Set that instance in the constructor by calling clock.New()
3. In this test, override the field by setting up a mock clock `keyStore.Clock = mockClock`. 
4. Assert that the time in the mock clock is the same as the `RevokedAt` time. 

This will need a little bit of refactoring of the `Storage.RevokeKey` method, but that's ok. The code will look much better.

Reply:
Thanks, updated.